scripts/build.py

```diff
@@ -0,0 +1,213 @@
+    Raises:
+        IOError: Current directory is not in a git repo
+
+    Returns:
+        (str): Root directory of git repo
+    """
+    return run_command(['git', 'rev-parse', '--show-toplevel'], collect_output=True)
+
+
+REPO_ROOT_DIR = get_repo_root_dir()
+
+def is_windows_os():
+    """Check if the current OS is Windows
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('win')
+
+
+def is_linux_os():
+    """Check if the current OS is Linux
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('linux')
+
+
+def is_mac_os():
+    """Check if the current OS is mac
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('darwin')
+
+
+def get_vcpkg_triplet(arch):
+    """ Get vcpkg target triplet (platform definition)
+
+    Args:
+        arch (str): Architecture (eg: 'x86', 'x64')
+
+    Raises:
+        ValueError: If current OS is not win,mac or linux
+    
+    Returns:
+        (str): Triplet name
+               Eg: "x64-windows-static"
+    """
+    triplet_name = [arch]
+    if is_windows_os():
+        triplet_name.append('windows')
+        # For windows, default is to build dynamic. Hence we specify static.
+        # For mac/linux, default is to build static libraries
+        triplet_name.append('static')
+    elif is_mac_os():
+        triplet_name.append('osx')
+    elif is_linux_os():
+        triplet_name.append('linux')
+    else:
+        raise ValueError('Unsupported operating system \
+                          We currently support windows, mac, linux')
+
+    return '-'.join(triplet_name)
+
+
+def step_cmake_configure(args):
+    """ CMake configure
+
+    If you are seeing problems when running this multiple times,
+    make sure to clean/delete previous build directory
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """   
+    cmd = ['cmake', '-S', '.', '-B', args.build_dir]
+    
+    # If generator is not specifed, default for platform is used by cmake, else 
+    # use the specified value
+    if args.generator:
+        cmd.extend(['-G', args.generator])
+    if args.config:
+        cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(args.config))
+    if args.build_tests:
+        cmd.append('-DFIREBASE_CPP_BUILD_TESTS=ON')
+        cmd.append('-DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON')
+    # If you want to avoid vcpkg altogether and have cpp dependencies installed elsewhere
+    if not args.no_vcpkg:
+        vcpkg_toolchain_file_path = os.path.join(REPO_ROOT_DIR, 'external', 'vcpkg', 
+                                                'scripts', 'buildsystems', 'vcpkg.cmake')
+        
+        cmd.append('-DCMAKE_TOOLCHAIN_FILE={0}'.format(vcpkg_toolchain_file_path))
+        cmd.append('-DVCPKG_TARGET_TRIPLET={0}'.format(get_vcpkg_triplet(args.arch)))
+    
+    cmd.append('-DFIREBASE_INCLUDE_FIRESTORE=OFF')
```

Comment:
Shouldn't this be a parameter we pass into the script?  Why is a lack of vkcpkg explicit to a lack of vcpkg, and why would a user know if/when Firestore is distabled?  Seems like a warning message on the console is warranted at the very least.

Reply:
Ignore the Firestore bit. That is a temporary placeholder for my test workflows to run without errors. Once we iron out the last few unit test errors, Firestore should be enabled by default.  I should have made a comment. 
re: vcpkg, this was placed to make the script more flexible incase uesrs didnt want to use vcpkg.